src/main/java/game/data/entity/specific/Villager.java

```diff
@@ -0,0 +1,141 @@
+package game.data.entity.specific;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+import game.data.coordinates.CoordinateDim3D;
+import game.data.entity.MobEntity;
+import game.data.entity.metadata.MetaData_1_13;
+import game.data.registries.RegistryManager;
+import game.data.villagers.VillagerTrade;
+import packets.DataTypeProvider;
+import se.llbit.nbt.ByteTag;
+import se.llbit.nbt.CompoundTag;
+import se.llbit.nbt.FloatTag;
+import se.llbit.nbt.IntTag;
+import se.llbit.nbt.ListTag;
+import se.llbit.nbt.StringTag;
+import se.llbit.nbt.Tag;
+
+/**
+ * Handle villagers because they have interesting metadata.
+ */
+public class Villager extends MobEntity {
+    private VillagerMetaData metaData;
+    private List<VillagerTrade> trades;
+    private int villagerLevel;
```

Comment:
I think this is unused currently, should this be written to NBT as well?

Reply:
The reason I didn't use the `villagerLevel` from the trade packet is because there appears to also be another villager level (`VillagerMetadata:122`) stored in `VillagerData.level` which is populated when the entity is added to the chunk.

I'm not entirely sure if these are both the same villager level and/or whether this needs updating. If so, then probably?

Reply:
As far as I'm aware, the villager's xp is a sufficient representation of the villager's "level" and it's almost as if the villager's level isn't even needed.